transport/src/main/java/io/netty/channel/socket/nio/NioServerSocketChannel.java

```diff
@@ -137,18 +152,34 @@ protected void doClose() throws Exception {
 
     @Override
     protected int doReadMessages(List<Object> buf) throws Exception {
-        SocketChannel ch = javaChannel().accept();
+        if (connectionCounter.incrementAndGet() > maxConnections) {
+            connectionCounter.decrementAndGet();
+            try {
+                //sleep a while so that we won't loop busily
+                Thread.sleep(10);
```

Comment:
"fail fast" do have bring us these benefits. but accept and then close may not suit for our case.
we just switched from tomcat to netty http and get troubled by the unlimited connections. 
we are expecting a feature same as "maxConnections" in tomcat, so that limited connections get processed simultaneously. 
each connection won't stay long(http request without keep-alive). so the backlog serve as a buffer, just that the connection in it need no fd.
accept and close usually means an error at client, and the client had no choice but retry and so cost extra performance.
accept and close is a way to solve the original problem. but in this way , as @normanmaurer  said below, no need to change netty, we may maintain the counter in our biz-logic ;)